src/component/StarRatings/StarRatings.ts

```diff
@@ -0,0 +1,134 @@
+import MovieInputStorage from "../../storage/MovieInputStorage";
+
+import starFilledImg from "../../image/star_filled.png";
+import emptyStarImg from "../../image/star_empty.png";
+
+import { $, createElement } from "../../utility/dom";
+import { StarRatingButton } from "./../StarRatingsButton/StarRatingsButton";
+import { STAR_RATINGS } from "./../../constant/starRatings.const";
+
+class StarRatings {
+  #movieId: number;
+  #starRatingsButtons: HTMLElement[];
+
+  constructor() {
+    this.#movieId = 0;
+    this.#starRatingsButtons = this.createStarRatingButtons();
+  }
+
+  setMovieId(movieId: number) {
+    this.#movieId = movieId;
+  }
+
+  getMovieInputData(movieId: number) {
+    return MovieInputStorage.getMovieInput(movieId);
+  }
+
+  initMyScoreSection(movieId: number) {
+    this.setAllStarRatingsButtonEmpty();
+
+    const movieInputData = this.getMovieInputData(movieId);
+    let starFilledCount = Number(movieInputData.starFilledCount) || 0;
+
+    this.setScoreDetailsElements(starFilledCount);
+    this.initStarRatingsButtons(starFilledCount);
+  }
+
+  initStarRatingsButtons(starFilledCount: number) {
+    this.#starRatingsButtons.forEach((starButton, index) => {
+      if (index < starFilledCount) {
+        const img = starButton.querySelector("img") as HTMLImageElement;
+        img.src = starFilledImg;
+      }
+    });
+  }
+
+  starRatingsButtonClickHandler(event: Event) {
+    const targetElement = event?.target;
+
+    if (targetElement instanceof Image) {
+      const starButtonId = Number(
+        targetElement.parentElement?.id.replace(/\D/g, "")
+      );
+      this.setStarRatingButtonsStatus(starButtonId + 1);
+    }
+  }
+
+  setScoreDetailsElements(starFilledCount: number) {
+    const myRatingsElement = $(".myscore-ratings") as HTMLElement;
+    const myRatingsDescriptionElement = $(
+      ".myscore-description"
+    ) as HTMLElement;
+
+    const score = starFilledCount * 2; // TODO: 단위 상수화
+    const description = STAR_RATINGS.scoreDescription[score];
+
+    myRatingsElement.textContent = `${score}`;
```

Comment:
사소한데 이게 더 깔끔하고 명시적인 느낌이에요. 템플릿 리터럴은 문자열 내에서 변수의 값을 동적으로 삽입하거나, 여러 줄에 걸쳐 문자열을 표현할 때 주로 사용하고, 형변환만 필요하면 형변환만 하는 방향으로요 :)

```suggestion
    myRatingsElement.textContent = String(score);
```